File: Entities/EmergencyCase.cs
namespace PrivateHospitalSystem.Entities
{
    public enum TriagePriority
    {
        NonUrgent = 1,
        Standard = 2,
        Urgent = 3,
        VeryUrgent = 4,
        Immediate = 5
    }

    public enum EmergencyStatus
    {
        Waiting,
        InProgress,
        Completed
    }

    public class EmergencyCase
    {
        public Guid Id { get; set; }

        public Guid PatientId { get; set; }
        public Patient? Patient { get; set; }

        public Guid? DoctorId { get; set; } 
        public Doctor? Doctor { get; set; }

        public string Complaint { get; set; } = string.Empty; 
        public TriagePriority Priority { get; set; } = TriagePriority.Standard;
        public EmergencyStatus Status { get; set; } = EmergencyStatus.Waiting;

        public DateTime ArrivedAt { get; set; } = DateTime.UtcNow;
        public DateTime? StartedAt { get; set; }
        public DateTime? CompletedAt { get; set; }
    }
}

File: frontend/src/types/emergencyCase.ts
export interface EmergencyCase {
  id: string
  patientId: string
  patientName: string
  doctorId: string | null
  doctorName: string | null
  complaint: string
  priority: string
  status: string
  arrivedAt: string
  startedAt: string | null
  completedAt: string | null
}

export interface CreateEmergencyCaseDto {
  patientId: string
  complaint: string
  priority: number
}

export interface CompleteEmergencyCaseDto {
  requiresAdmission: boolean
  bedId?: string
  admissionReason?: string
}